ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -161,8 +153,9 @@ void handleResponse(ResponseInfo responseInfo) {
     GetOperation getOperation = correlationIdToGetOperation.remove(getRequest.getCorrelationId());
     if (getOperations.contains(getOperation)) {
       getOperation.handleResponse(responseInfo);
-      if (getOperation.isOperationComplete() && getOperations.remove(getOperation)) {
-        onComplete(getOperation);
+      if (getOperation.isOperationComplete()) {
+        getOperations.remove(getOperation);
+        routerMetrics.operationDequeuingRate.mark();
```

Comment:
is it possible you will double count if `close()` has removed the operation already?